homeassistant/components/duotecno/config_flow.py

```diff
@@ -0,0 +1,67 @@
+"""Config flow for duotecno integration."""
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+from duotecno.controller import PyDuotecno
+from duotecno.exceptions import InvallidPassword
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_HOST, CONF_PASSWORD, CONF_PORT
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import FlowResult
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+STEP_USER_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_HOST): str,
+        vol.Required(CONF_PASSWORD): str,
+        vol.Required(CONF_PORT): int,
+    }
+)
+
+
+async def validate_input(hass: HomeAssistant, data: dict[str, Any]) -> dict[str, Any]:
+    """Validate the user input allows us to connect.
+
+    Data has the keys from STEP_USER_DATA_SCHEMA with values provided by the user.
+    """
+    # check if there is no duplicate
+    controller = PyDuotecno()
+    await controller.connect(
+        data[CONF_HOST], data[CONF_PORT], data[CONF_PASSWORD], True
+    )
+    return {"title": f"{data[CONF_HOST]}:{data[CONF_PORT]}"}
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for duotecno."""
+
+    VERSION = 1
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle the initial step."""
+        errors: dict[str, str] = {}
+        if user_input is not None:
+            try:
+                info = await validate_input(self.hass, user_input)
+            except ConnectionError:
+                errors["base"] = "cannot_connect"
+            except InvallidPassword:
+                errors["base"] = "invalid_auth"
+            except Exception:  # pylint: disable=broad-except
+                _LOGGER.exception("Unexpected exception")
+                errors["base"] = "unknown"
+            else:
+                return self.async_create_entry(title=info["title"], data=user_input)
```

Comment:
If you really want to go this route, you can do this:

```suggestion
                return self.async_create_entry(title=f"{data[CONF_HOST]}:{data[CONF_PORT]}", data=user_input)
```

However, I would not add host & port in the config entry title (as it is not so user friendly).

Alternatives:
- Only add the host
- Or, just use a default title